### PR TITLE
Add support for Team creation

### PR DESCRIPTION
**Pull request by mtscout6**
If this looks good to you guys I'll be following up with some team management capabilities as well in another PR.

### CODE REVIEW

**Comment by mtscout6** [comment]
What's the best way to go about testing these teams? It seems to me that you are doing everything against an Org that you actually plan to use. Do you have a test organization setup?

**Comment by mtscout6** [comment]
It looks like you are using the [github-api-tests](https://github.com/github-api-tests) org for testing this repo. In order to facilitate some better tests for Teams can you add either:

- add a team "Fixed Test Team" with the test user mikedeboertest
- add me to that Organization so I can set that up myself

**Comment by mtscout6** [comment]
Well that's frustrating, Lint only fails on CI but passes locally.... I'll get through it, just frustrating. Any thoughts on why that is? I'm running `npm run lint` locally.

**Comment by mtscout6** [comment]
Have you guys reduced the number of allowed concurrent builds in Travis? Only one job runs at a time and it's really slow to get through CI if there's issues to resolve.

**Comment by clayreimann** [comment]
I think OSS only gets one concurrent build.

**Comment by mtscout6** [comment]
Note that I added two other dev time features. I changed Travis to use container based infrastructure which should improve travis worker startup by about 20 seconds per job. I also added an ESLint plugin for mocha to ensure things like `it.only` or `describe.only` are not forgotten when committing.

**Comment by mtscout6** [comment]
https://blog.travis-ci.com/2014-07-18-per-repository-concurrency-setting/

I can't see what settings you have applied, but on a Saturday when OSS seems to slow down Travis should be allowing more concurrent jobs. Other OSS projects I've worked on have run concurrently.

**Comment by clayreimann** [comment]
I'll investigate. Don't think we've done anything special.

**Comment by mtscout6** [comment]
Either way I think everything should be wrapped up for this PR unless you guys see anything I may have missed.

**Comment by clayreimann** [comment]
As it happens we did limit concurrent jobs to 1. I've removed that setting

File: lib/Team.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
+/**
+ * @file
+ * @copyright  2013 Michael Aufreiter (Development Seed) and 2016 Yahoo Inc.
```
Comment (clayreimann) [inline lib/Team.js]: This probably wants to be you (or whomever you've signed your copyright over to), neither Yahoo nor Michael Aufreiter/Development Seed wrote this code.

File: lib/Team.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    * Add a member to the Team
+    * @see https://developer.github.com/v3/orgs/teams/#add-team-membership
+    * @param {string} username - can be one of: `all`, `maintainer`, or `member`
+    * @param {object} options - Parameters for adding a team member
+    * @param {string} [options.role=member] - The role that this user should have in the team. Can be one
+    * of: `member`, or `maintainer`
+    * @param {Requestable.callback} [cb] - will receive the membership status of added user
+    * @return {Promise} - the promise for the http request
+    */
+   addUser(username, options, cb) {
```
Comment (clayreimann) [inline lib/Team.js]: Do these want to be addMember/removeMember for consistency?

File: lib/Team.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+   /**
+    * Get repo management status for team
+    * @see https://developer.github.com/v3/orgs/teams/#remove-team-membership
+    * @param {string} owner - Organization name
+    * @param {string} repo - Repo name
+    * @param {Requestable.callback} [cb] - will receive the membership status of added user
+    * @return {Promise} - the promise for the http request
+    */
+   managesRepo(owner, repo, cb) {
```
Comment (clayreimann) [inline lib/Team.js]: `isManagedRepo`?

File: lib/Team.js [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    * @see https://developer.github.com/v3/orgs/teams/#add-or-update-team-repository
+    * @param {string} owner - Organization name
+    * @param {string} repo - Repo name
+    * @param {object} options - Parameters for adding or updating repo management for the team
+    * @param {string} [options.permission] - The permission to grant the team on this repository. Can be one
+    * of: `pull`, `push`, or `admin`
+    * @param {Requestable.callback} [cb] - will receive the membership status of added user
+    * @return {Promise} - the promise for the http request
+    */
+   addOrUpdateRepo(owner, repo, options, cb) {
```
Comment (clayreimann) [inline lib/Team.js]: `updateRepoManagement`?

File: lib/Team.js [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+   /**
+    * Remove repo management status for team
+    * @see https://developer.github.com/v3/orgs/teams/#remove-team-repository
+    * @param {string} owner - Organization name
+    * @param {string} repo - Repo name
+    * @param {Requestable.callback} [cb] - will receive the membership status of added user
+    * @return {Promise} - the promise for the http request
+    */
+   removeRepo(owner, repo, cb) {
```
Comment (clayreimann) [inline lib/Team.js]: `unmanageRepo`?

File: test/auth.spec.js [hunk 6]
```diff
@@ -51,7 +51,7 @@ describe('Github', function() {
                done();
             } catch (e) {
                try {
-                  if (err && err.request.headers['x-ratelimit-remaining'] === '0') {
+                  if (err && err.response.headers['x-ratelimit-remaining'] === '0') {
```
Comment (clayreimann) [inline test/auth.spec.js]: Good catch!

File: lib/Team.js (hunk 1 above)
Comment (mtscout6) [inline lib/Team.js]: Personally I don't see a need for any of that if you declare it for the repo as a whole. I'll change it to my name though.

**Comment by clayreimann** [comment]
I've made suggestions for some different method names, and will try and get a hold of the person who created the `github-api-tests` org to see about adding teams/some committers here as admins.

File: lib/Team.js (hunk 2 above)
Comment (mtscout6) [inline lib/Team.js]: I like that, I'll change it in just a minute.

File: lib/Team.js (hunk 4 above)
Comment (mtscout6) [inline lib/Team.js]: This one is tough, `update` gives a feeling that the repo is already managed, which may not be the case.

Comment (mtscout6) [inline lib/Team.js]: I do like adding `Management` to the end of the name though.

Comment (mtscout6) [inline lib/Team.js]: For consistency with the `unmanageRepo` name suggestion how does `manageRepo` sound?

File: lib/Team.js (hunk 2 above)
Comment (mtscout6) [inline lib/Team.js]: I just realized that I missed a test for the membership removal, working on that now.

File: lib/Team.js (hunk 3 above)
Comment (mtscout6) [inline lib/Team.js]: :+1:

**Comment by mtscout6** [comment]
I'm going to remove the `removeMembership` method. When trying to add a test for it I keep getting nothing but a `405 Not Allowed` error code. I'm not sure about the root cause of this, so I'm going to pull it out for now.

**Comment by mtscout6** [comment]
Well, I wonder if the membership removal is related to the current spikes happening within GitHub right now: https://status.github.com/

File: lib/Team.js (hunk 4 above)
Comment (clayreimann) [inline lib/Team.js]: Seems like a reasonable choice to me

**Comment by mtscout6** [comment]
I rebased this on the latest changes from master this morning, but for some reason Travis is not getting kicked off...

**Comment by clayreimann** [comment]
@mtscout6 Not to concerned, I assume they pass locally.

**Comment by clayreimann** [comment]
@mtscout6 v2.1 should be going out now

**Comment by mtscout6** [comment]
Thanks!

**Pull request merged**